libtcmu_log.c

```diff
@@ -359,15 +358,27 @@ static int create_stdout_output(int pri)
 
 static int create_file_output(int pri, const char *filename)
 {
-	int fd;
+	char log_file_path[PATH_MAX];
+	int fd, ret;
 
-	fd = open(filename, O_CREAT | O_APPEND | O_WRONLY, S_IRUSR | S_IWUSR);
-	if (fd < 0)
-		return  -1;
+	ret = tcmu_make_absolute_logfile(log_file_path, filename);
+	if (ret < 0) {
+		tcmu_err("tcmu_make_absolute_logfile failed\n");
+		return ret;
+	}
 
-	if (append_output(output_to_fd, close_fd, (void *)(intptr_t) fd,
-			  pri, TCMU_LOG_TO_FILE, filename) < 0)
-		return -1;
+	fd = open(log_file_path, O_CREAT | O_APPEND | O_WRONLY, S_IRUSR | S_IWUSR);
+	if (fd < 0) {
+		tcmu_err("Failed to open %s:%m\n", log_file_path);
+		return fd;
+	}
+
+	ret = append_output(output_to_fd, close_fd, (void *)(intptr_t) fd,
+			    pri, TCMU_LOG_TO_FILE, filename);
+	if (ret < 0) {
+		tcmu_err("Failed to append output file: %s\n", log_file_path);
+		return ret;
+	}
```

Comment:
Do you need a close() here?

Reply:
Just moved this codes from another place.

Yes and will fix it later.

Reply:
+1 Mike